### ISSUE TITLE
Not accessible to screen readers

### ISSUE THREAD

**Issue by rkingett** [sumatrapdfreader/sumatrapdf#321]
I am a blind user who is using the free screen reader NVDA. I would really love to use this program but the documents, even though they are text documents, will not read at all with NVDA.

Consider making this accessible in the future?

**Comment by mohammad-suliman** [comment]
yes, I agree, I believe that there are a blind people who want to use this program other than me and the person who reported this issue. this program focuses on simplicity, therefor, it is attractive for screen reader users. 
I realized that you intend to support narrator, which is a magnificent step forward. it will be also better if you consider supporting nvda - a free and open source screen reader, which is more advanced than narrator, and it is widely used among the blind community.

**Comment by rkingett** [comment]
I hope they do!

**Comment by SumatraPeter** [comment]
I don't know whether adding support for each screen reader requires dedicated code or not, but if we're talking of popularity and wide usage amongst the blind community then JAWS has to be supported IMO.

**Comment by rkingett** [comment]
One thing developers have noted in the past is that if it is fully
accessible to NVDA it will work with all others

Robert Kingett*, Journalist <https://blindjournalist.wordpress.com/resume/>*

*My Patreon* <https://www.patreon.com/blindwriter>

*Other ways to contact me*
<https://blindjournalist.wordpress.com/contact-this-blogger/>

**Comment by SumatraPeter** [comment]
> One thing developers have noted in the past is that if it is fully accessible to NVDA it will work with all others

If that's true then it'll certain make things easier for the SumatraPDF developers, because writing (and maintaining) custom code for each screen reader would be a nightmare.

**Comment by man0528** [comment]
I wish it would work with screen reader, just like FoxitReader.

**Comment by GitHubRulesOK** [comment]
Just to be clear the **current SumatraPDF build** can copy screen contents to clipboard and many speech apps **will READ out loud** the clipboard content

There are other apps that scrape the image pixels from the zoomed screen image and convert them to speech SumatraPDF is no different to any other input. It is the pixel quality that determines the value of the garbled translation result

The results of linking with NVDA & MSNarrator have varied over the years as either side of the links has changed it should be noted **ONLY the heading CONTROL FRAMEWORK or bookmarks to left are visible to screen narrators** the **rendered image area** that visual users see is _not accessible_ to screen readers

SumatraPDF is not customized to generally work with screen reader JAWS (Job Access With Speech) 
 " just like FoxitReader "

**Comment by rkingett** [comment]
Unfortunately, the developer does not want this to become accessible to NVDA, and similar. Unless there's a law we can use to force this reader to be accessible, it's gonna be a very tough fight. Also, copy and paste doesn't work. You should know that.and

**Comment by GitHubRulesOK** [comment]
@rkingett 
I think you misread the support info.
The developer **does** support and link to NVDA (on that help page) as well as mention Narrator but neither are able to read the surface textual image they cannot not "read out" the contents of a  comic book picture of words and the glyphs for text in a pdf are not always there to be copied to the system

The NVDA programmers should not be forced by law to use SumatraPDF as a screen image reader they have other issues to attend to.


I just re-tested NVDA and could use it to fully read the PDF Title and SumatraPDF pull down controls without any problem

**Comment by SumatraPeter** [comment]
Let's ignore image-based PDFs for the moment. What I want to know is, what is _technically_ preventing screen readers from being able to access the content of text-based PDFs (and possibly other supported ebook formats) displayed by Sumatra? Hopefully @kjk can elaborate on what the problem is, and whether it would require massive rework to be able to improve matters in terms of accessibility.

**Comment by kjk** [comment]
For standard windows controls (like buttons or menus) Windows provides the necessary code to support accessibility features used by screen readers etc.
Everything else (like PDF text rendered on screen) needs custom code. This code is not easy to write and there is a lot of other code that needs to be written.

**Comment by man0528** [comment]
> Unfortunately, the developer does not want this to become accessible to NVDA, and similar. Unless there's a law we can use to force this reader to be accessible, it's gonna be a very tough fight. Also, copy and paste doesn't work. You should know that.and


Copying content to the clipboard to make screen readers read aloud would be a very bad experience. Visually impaired users do not only need to know the pure text content, rich formatting information is more useful.

**Comment by GitHubRulesOK** [comment]
@man0528 
I agree scraping scanned text needs dedicated software to indicate emphasis such as quoted / braced / italics / emboldened / color / tabular contents etc. and many discussions abound around how that audio descriptive content could detract from a plain audio stream.
I have no idea how the commercial products such as OpenBooks deal with that aspect. I am simply suggesting that TTS screen scraping via the clipboard is better than "nothing"

**Comment by SumatraPeter** [comment]
> Everything else (like PDF text rendered on screen) needs custom code. This code is not easy to write and there is a lot of other code that needs to be written.

Unless the code has to be written at the Sumatra level, clearly it's not a high priority even for a large organization like Artifex that is responsible for MuPDF.

**Comment by GitHubRulesOK** [comment]
On Windows the inbuilt Screen Reader is MS Edge (currently 2024 is vIII+) which does a very good job working with both PDF and  ePub plus many others but only PDF has good Natural Reading.

Here reading this Page as a PDF.

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/d5e81fe3-bc90-47ba-b344-3ad769afd5dd)

However it does not attempt ePub
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/0a6a8176-73a0-44ec-9563-3fde46f55060)

A very capable audio reader is Balabolka which converts many file types to text and can read them different ways.

I have explained many times SumatraPDF can easily send the filename and or page number to any suitable powerful external reader via command line. Edge TTSreader Balabolka and many others can be used to do convert documents into text for TTS.

There are also many screen scrapers that can fill the gap between pixel render and speech by use onscreen OCR.

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/f3ac8a56-d0df-4dae-94b4-54788a8b5952)

**Comment by cary-rowen** [comment]
@GitHubRulesOK 
> On Windows the inbuilt Screen Reader is MS Edge (currently 2024 is vIII+) which does a very good job working with both PDF and ePub plus many others but only PDF has good Natural Reading.
I would like to correct that MS Edge is not a built-in screen reader in Windows.
It’s just that the browser has a feature that reads web pages or PDF text aloud.
Windows' built-in screen reader is called Narrator, and you can turn it on or off by pressing Windows+Ctrl+Enter.

I don't think the reporter's request was for the project to support reading pdf documents consisting entirely of images.

@rkingett  wrote:
> I am a blind user who is using the free screen reader NVDA. I would really love to use this program but the documents, even though they are text documents, will not read at all with NVDA.



I haven't seen any accessibility efforts from sumatrapdf in all these years

**Comment by GitHubRulesOK** [comment]
@cary-rowen 
Not looking for a bun fight I accept your correction that Windows Screen Reader is Narrator, but AFAIK it can not access PDF data.

Raw PDF text is NUMERIC (often out of human order) and needs conversion into a semblance of text strings.

However the Windows PDF reader is MS Edge and in the past would take PDF reader control away from any other 3rd party!
I have not tested NVDA or narrator in Edge to give a comparison, perhaps I should.

Acrobat has a built in heuristics based text extractor, designed especially for extracting text from PDF into a different accessible "reflow" format than is native inside most PDFs. You can "save as" such output as a text file.  So does not attempt any document formats other than their own PDF [/A] XFA/XMP or otherwise restructured PDFs Much of that code comes from Acrobat editor (crippled in reader DC) and causes much of the slow/bloat.

**Comment by GitHubRulesOK** [comment]
Just an update that SumatraPDF now supports read Text as Speech where possible. Thus, there will be many exceptions, but most document formats can be "Heard Aloud"
This can be a selected area of text or a "Whole Page" with any User based Microsoft Windows Voices There are always going to be files or parts of pages that cannot be read or with a PDF sound totally "Garbled". But if the file has logical text (and many PDFs do NOT) then it will be read in line order. Do not be surprised that PDF pages can be written in a totally random order but LOOK like they may work.

 Basically, if the words sound "out of order", then most likely that is the way the PDF writer charted them. As PDF is a Page "DRAWING" system, using character "tokens" for calling a drawn font. (the "token" can be a totally different sound)

<img width="788" height="247" alt="Image" src="https://github.com/user-attachments/assets/4fafc451-a767-415e-b331-cf6fb9c4f2e4" />

**Comment by GitHubRulesOK** [comment]
For others landing here the current situation should be as described in [https://www.sumatrapdfreader.org/docs/Accessibility](https://www.sumatrapdfreader.org/docs/Accessibility)
> Microsoft Narrator[ #](https://www.sumatrapdfreader.org/docs/Accessibility#microsoft-narrator)
Start Narrator (Windows built-in).
Open a document in SumatraPDF.
Select text — Narrator should read the selection.
Supported document types for UIA: PDF, XPS, DjVu.
Known issue: Narrator sometimes stops reading until you switch focus to another window and back.

TTS has its own page at https://www.sumatrapdfreader.org/docs/Read-Aloud-TTS

**Comment by cary-rowen** [comment]
Hi @GitHubRulesOK 
I tested version 3.6 with Narrator and NVDA. I can only access the table of contents. As for the main body text, I can focus on it, but screen readers are completely unable to read it. What should I do?

**Comment by kjk** [comment]
try latest [pre-release](https://www.sumatrapdfreader.org/prerelease) as this is where I've made improvements. If things are missing, please open a new issue and describe in detail what is missing and how to verify it's fixed

**Comment by cary-rowen** [comment]
I tried the pre-release version, and the situation was the same: only the directory was accessible, the main content was completely inaccessible.

64-bit
Pre-release
SumatraPDF
build
Built: Jul 28 2026 22:58:30

**Comment by kjk** [comment]
I would need more information:
* what exactly are you doing
* what do you expect to happen
* what happens

**Comment by GitHubRulesOK** [comment]
@cary-rowen 
Narrator can be limited in its access so here I know why one line can be heard when entered but neither when  not entered so narrator will say "slash 200" and that is that forms/annotation are not normal body text
in the second case I must select the text for narrator to read it.
<img width="895" height="467" alt="Image" src="https://github.com/user-attachments/assets/294f6e81-ba9b-4769-b316-8b251d59c9d4" />

<img width="1141" height="277" alt="Image" src="https://github.com/user-attachments/assets/48b6bc1a-569b-43be-aa7a-61c09b3ee8bc" />
